src/zarr/core/array_spec.py

```diff
@@ -29,7 +29,7 @@ def __init__(
         prototype: BufferPrototype,
     ) -> None:
         shape_parsed = parse_shapelike(shape)
-        dtype_parsed = parse_dtype(dtype)
+        dtype_parsed = dtype  # parsing is likely not needed here
```

Comment:
```suggestion
        dtype_parsed = dtype
```

The input type is already `np.dtype[Any]` so I don't think we need to parse this.